src/Services/ActivationService.cs

```diff
@@ -85,6 +87,18 @@ private async Task StartupAsync()
         // Subscribe to theme changes.
         _themeSelectorService.ThemeChanged += (_, theme) => App.MainWindow.SetRequestedTheme(theme);
         _themeSelectorService.SetRequestedTheme();
+
+        // Disable Hyper-V extension on ARM on first run
+        if (!await _localSettingsService.ReadSettingAsync<bool>(WellKnownSettingsKeys.IsNotFirstRun))
+        {
+            var arch = RuntimeInformation.ProcessArchitecture;
+            if (arch == Architecture.Arm64 || arch == Architecture.Arm || arch == Architecture.Armv6)
+            {
+                await _localSettingsService.SaveSettingAsync<bool>(CommonConstants.HyperVUniqueId + "-ExtensionDisabled", true);

```

Comment:
Why is this a setting, as opposed to some sort of "IsSupported" check elsewhere in the feature?

Reply:
Changed the implementation completely.